### PR TITLE
symmetric: Handle empty IVs with AES-GCM

### LINKED ISSUE #83: Non-informative message when building GCM decryptor without an IV
Following code, note setting of an empty IV vector:
```c++
    auto decryptor = mococrw::AESCipherBuilder{mococrw::SymmetricCipherMode::GCM,
                                               mococrw::SymmetricCipherKeySize::S_128,
                                               mKey}
                         .setPadding(mococrw::SymmetricCipherPadding::NO)
                         .setIV({})
                         .buildAuthenticatedDecryptor();
```
Results in an exception:
```
terminate called after throwing an instance of 'mococrw::openssl::OpenSSLException'
  what():  error:00000000:lib(0):func(0):reason(0): 0
```
Which has no useful information for a developer to understand the issue. Maybe this can be improved.

### PR DESCRIPTION
**Pull request by neverpanic**
OpenSSL does not support using an empty IV with AES-GCM, but the error message is most unhelpful. Improve this by checking for this situation and throwing a MoCOCrWException with a clearer message.

Closes: #83.

### CODE REVIEW

**Comment by neverpanic** [comment]
The failed push job is a red herring, please ignore that.

**Review by garw — APPROVED** [review]

**Comment by neverpanic** [comment]
I'll wait for the internal CI to pass before merging this.

**Pull request merged**
